core/audits/bf-cache.js

```diff
@@ -0,0 +1,126 @@
+  actionableFailureType: 'Actionable',
+  /** Back/forward cache failure type for a back/forward cache failure that the user cannot address themselves. Shown in a table column with other failure types. */
+  notActionableFailureType: 'Not actionable',
+  /** Back/forward cache failure type for a back/forward cache failure caused by missing browser support. Shown in a table column with other failure types. */
+  supportPendingFailureType: 'Pending browser support',
+  /** Label for a column in a data table; entries in the column will be a string representing a reason the back/forward cache could not be used. */
+  failureReasonColumn: 'Failure reason',
+  /** Label for a column in a data table; entries in the column will be a string representing the type of failure preventing the back/forward cache from being used. */
+  failureTypeColumn: 'Failure type',
+  /**
+   * @description [ICU Syntax] Label for an audit identifying the number of back/forward cache failure reasons found in the page that the user can address.
+   */
+  displayValue: `{itemCount, plural,
+    =1 {1 actionable failure reason}
+    other {# actionable failure reasons}
+    }`,
+};
+/* eslint-enable max-len */
+
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+/** @type {LH.Crdp.Page.BackForwardCacheNotRestoredReasonType[]} */
+const ORDERED_FAILURE_TYPES = ['PageSupportNeeded', 'Circumstantial', 'SupportPending'];
+
+/** @type {Record<LH.Crdp.Page.BackForwardCacheNotRestoredReasonType, string | LH.IcuMessage>} */
+const FAILURE_TYPE_TO_STRING = {
+  PageSupportNeeded: str_(UIStrings.actionableFailureType),
+  Circumstantial: str_(UIStrings.notActionableFailureType),
+  SupportPending: str_(UIStrings.supportPendingFailureType),
+};
+
+class BFCache extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'bf-cache',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      supportedModes: ['navigation', 'timespan'],
+      requiredArtifacts: ['BFCacheFailures'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
+    const failures = artifacts.BFCacheFailures;
+    if (!failures.length) return {score: 1};
+
+    // TODO: Analyze more than one bf cache failure.
+    const {notRestoredReasonsTree} = failures[0];
+
+    /** @type {LH.Audit.Details.TableItem[]} */
+    const results = [];
+    let numActionable = 0;
+
+    for (const failureType of ORDERED_FAILURE_TYPES) {
+      const reasonsMap = notRestoredReasonsTree[failureType];
+
+      // https://github.com/Microsoft/TypeScript/issues/12870
+      const reasons = /** @type {LH.Crdp.Page.BackForwardCacheNotRestoredReason[]} */
+        (Object.keys(reasonsMap));
```

Comment:
The error is correct, and in our case probably a good warning that the strings and protocol responses may get out of sync (which is already handled by the `?? reason` fallback below).

`for (const [reason, frameUrls] of Object.entries(reasonsMap)) {`
and switching `NotRestoredReasonDescription` to type `Record<string, {name: IcuMessage}>` would make that explicit and not require a type cast